src/lib/pickles/impls.ml

```diff
@@ -216,9 +216,28 @@ module Wrap = struct
     let to_bits x = Field.unpack x ~length:Field.size_in_bits
   end
 
-  let input () =
+  let input ~feature_flags () =
     let lookup =
-      { Types.Wrap.Lookup_parameters.use = No
+      { Types.Wrap.Lookup_parameters.use =
+          (let { Plonk_types.Features.range_check0
+               ; range_check1
+               ; foreign_field_add = _ (* Doesn't use lookup *)
+               ; foreign_field_mul
+               ; xor
+               ; rot
+               ; lookup
+               ; runtime_tables = _ (* Fixme *)
+               } =
+             feature_flags
+           in
+           Array.reduce_exn ~f:Plonk_types.Opt.Flag.( ||| )
+             [| range_check0
+              ; range_check1
+              ; foreign_field_mul
+              ; xor
+              ; rot
+              ; lookup
+             |] )
```

Comment:
[nit] Alternatively, we could forego creating a new array
```suggestion
         range_check0 |||  range_check1 ||| foreign_field_mul ||| xor ||| rot ||| lookup
```
But maybe there's some kind of error that we might be able to catch more easily in the current form?